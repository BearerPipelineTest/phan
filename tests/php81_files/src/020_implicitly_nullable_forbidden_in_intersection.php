<?php

function example20(Countable&Traversable $ct = null): int {
    foreach ($ct as $x) {
        var_dump($x);
    }
    return count($ct);
}
example20(null);
example20(new ArrayObject([]));
