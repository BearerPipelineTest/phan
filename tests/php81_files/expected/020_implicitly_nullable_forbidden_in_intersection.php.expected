%s:7 PhanTypeMismatchArgumentNullableInternal Argument 1 ($value) is $ct of type ?\Countable|?\Traversable|?iterable but \count() takes \Countable|\ResourceBundle|\SimpleXMLElement|array (expected type to be non-nullable)
